packages/quicktype-core/src/language/CPlusPlus.ts

```diff
@@ -521,6 +516,115 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
         this.setupGlobalNames();
     }
 
+    // union typeguard
+    isUnion(t: Type | UnionType): t is UnionType {
+        return t.kind === "union";
+    }
+
+    // Returns true if the type can be stored in
+    // a stack based optional type. This requires
+    // that the type does not require forward declaration.
+    isOptionalAsValuePossible(t: Type): boolean {
+        if (this.isForwardDeclaredType(t)) return false;
+
+        if (this.isUnion(t)) {
+            // There is something stinky about this test.
+            // There is special handling somewhere that if you
+            // have the following schema
+            // {
+            //     "$schema": "http://json-schema.org/draft-06/schema#",
+            //     "$ref": "#/definitions/List",
+            //     "definitions": {
+            //         "List": {
+            //             "type": "object",
+            //             "additionalProperties": false,
+            //             "properties": {
+            //                 "data": {
+            //                     "type": "string"
+            //                 },
+            //                 "next": {
+            //                     "anyOf": [
+            //                         {
+            //                             "$ref": "#/definitions/List"
+            //                         }
+            //                         {
+            //                             "type": "null"
+            //                         }
+            //                     ]
+            //                 }
+            //             },
+            //             "required": [],
+            //             "title": "List"
+            //         }
+            //     }
+            // }
+            // Then a variant is not output but the single item inlined
+            //
+            //     struct TopLevel {
+            //       std::optional<std::string> data;
+            //       std::optional<TopLevel> next;
+            //     };
+            //
+            // instead of
+            //     struct TopLevel {
+            //       std::optional<std::string> data;
+            //       std::shared_ptr<TopLevel> next;
+            //     };
+            //
+            // checking to see if the collapse of the variant has
+            // occured and then doing the isCycleBreakerType check
+            // on the single type the variant would contain seems
+            // to solve the problem. But does this point to a problem
+            // with the core library or with the CPlusPlus package
+            const [_hasNull, nonNulls] = removeNullFromUnion(t);
```

Comment:
I think you can write `const [, nonNulls] = ...`

Reply:
Ok. I'll fix that.

Reply:
should be fixed now.